src/gordon_gcp/plugins/service/gdns_publisher.py

```diff
@@ -195,116 +183,76 @@ def _format_resource_record_changes(self, action, resource_record):
             action: [resource_record]
         }
 
-    async def _publish_changes(self, changes, base_changes_url):
-        """Publish changes to the Google DNS Changes API.
-
-        Args:
-            changes (dict): The DNS record changes to publish.
-            base_changes_url (str): The URL to use to publish changes.
-
-        Returns:
-            change_id (str): Change ID of the Changes resource.
-        """
-        resp = await self.http_client.request(
-            'post', base_changes_url, json=changes)
-        resp_dict = json.loads(resp)
-
-        # TODO: create another task to measure propagation time
-        return resp_dict['id']
-
-    async def _handle_additions_conflict(self, changes, resource_records_url):
-        """Compensate for an additions Changes request that returns an HTTP
-           409.
-
-            WARNING: This function fetches _all_ records in a given Managed
-            Zone and may be slow if the zone contains many records.
+    async def _get_conflicting_rrsets(self, zone, changes):
+        """Get the GDNS version of records that have an addition conflict.
 
         Args:
+            zone (str): The DNS zone name.
             changes (dict): Changes containing an addition conflict.
-            resource_records_url (str): The URL to use to get the zone's
-                resource records.
 
         Returns:
-            dict containing original additions plus deletions.
+            list of rrsets: The GDNS version of the records with the conflict.
```

Comment:
what does `GDNS version` mean here?

Reply:
There's the version we're trying to add, and the version GDNS has.  Version meaning same keys, different values.  I'm trying to come up with a better way to put it, but having difficulty.

Reply:
'get the GDNS data for records with a conflict'?

Reply:
"""Get conflicting records as reflected in GDNS.
            (list[rrset]): The conflicting records as reflected in GDNS.

How's that?

Reply:
Ok, I refactored.  @econchick ?

Reply:
:+1: